azure-ipam/ipconfig/ipconfig.go

```diff
@@ -39,18 +39,24 @@ func CreateIPConfigReq(args *cniSkel.CmdArgs) (cns.IPConfigRequest, error) {
 }
 
 // ProcessIPConfigResp processes the IPConfigResponse from the CNS.
-func ProcessIPConfigResp(resp *cns.IPConfigResponse) (*netip.Prefix, error) {
-	podCIDR := fmt.Sprintf(
-		"%s/%d",
-		resp.PodIpInfo.PodIPConfig.IPAddress,
-		resp.PodIpInfo.NetworkContainerPrimaryIPConfig.IPSubnet.PrefixLength,
-	)
-	podIPNet, err := netip.ParsePrefix(podCIDR)
-	if err != nil {
-		return nil, errors.Wrapf(err, "cns returned invalid pod CIDR %q", podCIDR)
+func ProcessIPConfigResp(resp *cns.IPConfigResponse) (*[]netip.Prefix, error) {
+	var podIPNets []netip.Prefix
+	podIPNets = make([]netip.Prefix, len(resp.PodIpInfo))
+
+	for i := range resp.PodIpInfo {
+		podCIDR := fmt.Sprintf(
+			"%s/%d",
+			resp.PodIpInfo[i].PodIPConfig.IPAddress,
+			resp.PodIpInfo[i].NetworkContainerPrimaryIPConfig.IPSubnet.PrefixLength,
+		)
+		podIPNet, err := netip.ParsePrefix(podCIDR)
+		podIPNets[i] = podIPNet
```

Comment:
there is no point in assigning this if err is non-nil. This should come after line 56